lib/taskana-core/src/test/java/acceptance/task/QueryTaskWithAttachment.java

```diff
@@ -74,24 +74,24 @@ public void testIfNewTaskHasEmptyAttachmentList() {
     }
 
     @WithAccessId(
-            userName = "user_1_1",
-            groupNames = {"group_1"})
+        userName = "user_1_1",
+        groupNames = {"group_1"})
     @Test
     public void testIfAttachmentSummariesAreCorrect()
-            throws InvalidArgumentException, TaskNotFoundException, NotAuthorizedException {
+        throws InvalidArgumentException, TaskNotFoundException, NotAuthorizedException {
         TaskService taskService = taskanaEngine.getTaskService();
         // find Task with ID TKI:00...00
         List<TaskSummary> tasks = taskService.createTaskQuery()
-                .classificationKeyIn("T2000")
-                .customAttributeIn("1", "custom1")
-                .list();
+            .classificationKeyIn("T2000")
+            .customAttributeIn("1", "custom1")
+            .list();
         assertEquals(1, tasks.size());
 
         Task originalTask = taskService.getTask("TKI:000000000000000000000000000000000000");
         Attachment originalAttachment = originalTask.getAttachments().get(0);
 
         // Test if it's the Summary of the Original Attachment
-        AttachmentSummary queryAttachmentSummary = tasks.get(0).getAttachmentSummaries().get(0);
+        AttachmentSummary queryAttachmentSummary = tasks.get(0).getAttachmentSummaries().get(1);
```

Comment:
this looks not safe to me. Maybe we should explicitly compare the attechments with the same id or some other match code? 
Why is the order different in the two case and element 0 in one call is element 1 in the other?

Reply:
Distinct can change the order of the query.